server/src/api/tamagotchiCare.ts

```diff
@@ -0,0 +1,36 @@
+import { AxiosResponse } from 'axios';
+
+import api from '../shared/api';
+import { Tamagotchi } from '../types/Tamagotchi';
+
+interface TamagotchiCareOptions {
+  clean?: Tamagotchi['clean'];
+  entertainment?: Tamagotchi['entertainment'];
+  food?: Tamagotchi['food'];
+  id: Tamagotchi['id'];
+}
+
+type TamagotchiCareData = Pick<
+  TamagotchiCareOptions,
+  'clean' | 'entertainment' | 'food'
+>;
+
+const tamagotchiCare = (options: TamagotchiCareOptions) => {
+  const data: TamagotchiCareData = (
+    Object.keys(options) as Array<keyof typeof options>
+  ).reduce((accumulator, key) => {
+    if (!['clean', 'entertainment', 'food'].includes(key)) {
+      return accumulator;
+    }
+    if (!options[key]) {
+      return accumulator;
+    }
+    return { ...accumulator, [key]: options[key] };
+  }, {});
```

Comment:
This is so hard to read. The following takes more passes but is honestly way more readable:

```suggestion
const data = Object.keys(options)
  .map((key) => ([key, options[key]]))
  .filter(([, value]) => value !== undefined)
  .filter(([key]) => ['clean', 'entertainment', 'food'].includes(key))
  .reduce((accumulator, [key, value]) => ({ ...accumulator, [key]: value }), {});
```